client/app/models/rwcds-form.js

```diff
@@ -133,4 +133,23 @@ export default class RwcdsFormModel extends Model {
   @attr('date') overriddencreatedon;
 
   @attr('number') utcconversiontimezonecode;
+
+  async save() {
+    await this.saveDirtyAffectedZoningResolutions();
+    await super.save();
+  }
+
+  async saveDirtyAffectedZoningResolutions() {
+    return Promise.all(
+      this.affectedZoningResolutions
+        .filter((zoningResolution) => zoningResolution.hasDirtyAttributes)
+        .map((zoningResolution) => zoningResolution.save()),
+    );
+  }
+
+  get isAffectedZoningResolutionsDirty() {
+    const dirtyZrs = this.affectedZoningResolutions.filter((zr) => zr.hasDirtyAttributes);
+
+    return dirtyZrs.length > 0;
+  }
```

Comment:
Great. I love how this is all split out into named methods. Easy to follow and reason about. Great job.